### PR TITLE
[REVIEW] Remove ProjectRemoveRule

### LINKED ISSUE #1378: [BUG] Renaming cols fails hen selecting all cols
Renaming cols fails hen selecting all cols

**Works as Expected**
```
import cudf
df = cudf.DataFrame()
df["_col0"] = range(2)
df["_col1"] = range(2)

bc.create_table('t', df)
bc.sql('select _col0 as a from t').compute()
```
Expected Output:
```
	a
0	0
1	1
```

**Failure Case:**
```
import cudf
df = cudf.DataFrame()
df["_col0"] = range(2)
df["_col1"] = range(2)

bc.create_table('t', df)
bc.sql('select _col0 as a, _col1 as b from t').compute()
```
Wrong Output
```	
    _col0 _col1
0	0	0
1	1	1
```

**Reason:**
This happens only when selecting all columns, when selecting all columns, calcite is optimizing the algebra to select the full table and skipping rename functions.
```
DEBUG: com.blazingdb.calcite.application.RelationalAlgebraGenerator - non optimized
LogicalProject(a=[$0], b=[$1])
  LogicalTableScan(table=[[main, t]])

DEBUG: com.blazingdb.calcite.application.RelationalAlgebraGenerator - optimized
LogicalTableScan(table=[[main, t]])
```

### PR DESCRIPTION
**Pull request by Christian8491**
This PR closes #1378

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: engine/src/execution_graph/logic_controllers/BatchProcessing.cpp [hunk 1]
```diff
@@ -346,15 +346,29 @@ kstatus Projection::run() {
     CodeTimer timer;
 
     std::unique_ptr <ral::cache::CacheData> cache_data = this->input_cache()->pullCacheData();
-    while(cache_data != nullptr ){
-        std::vector<std::unique_ptr <ral::cache::CacheData> > inputs;
-        inputs.push_back(std::move(cache_data));
+    std::vector<std::string> data_names = cache_data->names();
```
Comment (williamBlazing) [inline engine/src/execution_graph/logic_controllers/BatchProcessing.cpp:349]: You need to null check cache_data first

Comment (williamBlazing) [reply engine/src/execution_graph/logic_controllers/BatchProcessing.cpp:349]: also rename `data_names` to `column_names`

Comment (williamBlazing) [reply engine/src/execution_graph/logic_controllers/BatchProcessing.cpp:349]: hmmm, the more i think about it, it should never be null. But to be safe, add something like `RAL_EXPECTS(cache_data != nullptr, "ERROR: Projection::run() first input CacheData was nullptr")`

File: engine/src/execution_graph/logic_controllers/BatchProcessing.cpp [hunk 2]
```diff
@@ -346,15 +346,29 @@ kstatus Projection::run() {
...
 
-        ral::execution::executor::get_instance()->add_task(
-                std::move(inputs),
-                this->output_cache(),
-                this);
+    // When this kernel will project all the columns (with or without aliases)
+    // we want to avoid caching and decahing for this kernel
+    bool by_passing_project, by_passing_project_with_aliases;
```
Comment (williamBlazing) [inline engine/src/execution_graph/logic_controllers/BatchProcessing.cpp:353]: bypass or bypassing is one word, not two words. So these variables should be bypassing_project, bypassing_project_with_aliases

Comment (williamBlazing) [reply engine/src/execution_graph/logic_controllers/BatchProcessing.cpp:353]: byPassingProject should be bypassingProject

**Review by williamBlazing — CHANGES_REQUESTED** [review]

**Pull request merged**